File: market/telemetry/telemetry.go
// Package telemetry defines infrastructure-neutral market calculation events.
package telemetry

import "time"

// Event is emitted at an authoritative calculation service boundary.
type Event struct {
	Operation     string
	DailyTideID   string
	Sector        string
	MappingID     string
	Status        string
	CoveredWeight int64
	Checksum      string
	Duration      time.Duration
	Err           error
}

// Recorder can bridge calculation events to structured logs and metrics.
type Recorder interface {
	RecordMarketCalculation(Event)
}

// NopRecorder is the default recorder when observability is not configured.
type NopRecorder struct{}

func (NopRecorder) RecordMarketCalculation(Event) {}

File: server/src/market/telemetry.ts
/** Infrastructure-neutral market calculation telemetry. */

export interface MarketCalculationEvent {
  operation: string;
  dailyTideId: string;
  sector: string;
  mappingId: string;
  status: string;
  coveredWeight: number;
  checksum: string;
  durationMs: number;
  error?: unknown;
}

export interface MarketRecorder {
  recordMarketCalculation(event: MarketCalculationEvent): void;
}

export class NopMarketRecorder implements MarketRecorder {
  recordMarketCalculation(_event: MarketCalculationEvent): void {
    // Intentionally empty when observability is not configured.
  }
}
